### PR TITLE
[Destination-MotherDuck][fix]: Don't lose Platform-generated `id` in state messages

### PR DESCRIPTION
**Pull request by aaronsteers**
## What

1. Adds "Patched" versions of the data classes and serialization classes. 
2. Overrides just enough of the `Destination` base class implementation to use the new classes.

## Can this PR be safely reverted and rolled back?

- [ ] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-integrations/connectors/destination-motherduck/destination_motherduck/destination.py [hunk 1]
```diff
@@ -29,6 +45,29 @@
...
+@dataclass
+class PatchedAirbyteStateMessage(AirbyteStateMessage):
+    """Declare the `id` attribute that platform sends."""
+
+    id: int | None = None
+    """Injected by the platform."""
+
+
+@dataclass
+class HackedAirbyteMessage(AirbyteMessage):
```
Comment (guenp) [inline airbyte-integrations/connectors/destination-motherduck/destination_motherduck/destination.py:57]:
```suggestion
class PatchedAirbyteMessage(AirbyteMessage):
```

File: airbyte-integrations/connectors/destination-motherduck/destination_motherduck/destination.py [hunk 2]
```diff
@@ -29,6 +45,29 @@
...
+    """Keep all defaults but override the type used in `state`."""
+
+    state: PatchedAirbyteStateMessage | None = None
+    """Override class for the state message only."""
+
+
+PatchedAirbyteMessageSerializer: type = Serializer(
+    HackedAirbyteMessage,
```
Comment (guenp) [inline airbyte-integrations/connectors/destination-motherduck/destination_motherduck/destination.py:65]:
```suggestion
    PatchedAirbyteMessage,
```

File: airbyte-integrations/connectors/destination-motherduck/destination_motherduck/destination.py [hunk 3]
```diff
@@ -23,12 +39,36 @@
 from destination_motherduck.processors.duckdb import DuckDBConfig, DuckDBSqlProcessor
 from destination_motherduck.processors.motherduck import MotherDuckConfig, MotherDuckSqlProcessor
 
+
```
Comment (aaronsteers) [inline airbyte-integrations/connectors/destination-motherduck/destination_motherduck/destination.py:42]:
```suggestion
```

**Comment by aaronsteers** [comment]
/bump-version type=patch changelog="Fix: Preserve Platform-generated id in state messages".

**Comment by aaronsteers** [comment]
/bump-version type=patch changelog="Fix: Preserve Platform-generated id in state messages."

**Comment by aaronsteers** [comment]
/bump-version type=patch changelog="Preserve Platform-generated id in state messages."

**Comment by aaronsteers** [comment]
/bump-version type=patch changelog="Preserve Platform-generated id in state messages."

**Comment by michrzan** [comment]
Thanks @aaronsteers ! This solved the NullPointerException I was getting earlier. Now, however, I am getting this one:

`Error when executing SQL: -- Write from PyArrow table INSERT INTO airbyte."my_table_01jbf225j" ("id", "name", "_airbyte_raw_id", "_airbyte_extracted_at", "_airbyte_meta") SELECT "id", "name", "_airbyte_raw_id", "_airbyte_extracted_at", "_airbyte_meta" FROM pa_table 
ProgrammingError(duckdb.duckdb.CatalogException) Catalog Error: Table with name pa_table does not exist! Did you mean "pg_tables"?`

is the pa_table not templated correctly in the SQL string?

**Comment by aaronsteers** [comment]
@michrzan - We are seeing the same and expecting to roll out a fix later today.

**Comment by aran159** [comment]
Hi @aaronsteers,

Thanks a lot for this fix — it was really helpful! 🙏

I’m currently developing a custom Salesforce destination using airbyte-cdk==6.60.0 on Airbyte Platform v7.2.0, and this resolved an issue I was running into:

```
State message does not contain id
```

If you don’t mind, I’d love to better understand the root cause. I’m still getting up to speed with the internals of Airbyte, so apologies in advance if these are basic questions:

1.	When did this issue start happening? Was it introduced by any recent changes in the Airbyte Protocol or CDK?
2.	Does this happen with all source connectors, or is it unrelated to the source side (since sources are the ones emitting the state messages)?
3. Why do other destinations written with python CDK 6.x, like destination-sqlite do not seem to be affected by this issue?

Thanks again for the fix and all the work behind it!

**Pull request merged**